.github/workflows/rust.yml

```diff
@@ -16,7 +16,42 @@ jobs:
       - uses: ./.github/actions/asdf
         with:
           rust: true
-      - run: cargo check
-        working-directory: bindings/rust
+      - name: Cache cargo registry
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-registry-
+      - name: Cache cargo target
+        uses: actions/cache@v2
+        with:
+          path: target
+          key: ${{ runner.os }}-cargo-target-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-target-
       - run: cargo check
         working-directory: cmd/tests/reprolang
+  rust-bindings-test:
```

Comment:
Could we directly use https://github.com/Swatinem/rust-cache instead of having this custom code with key-specificity etc here?